### PR TITLE
SCB-315 add common AddressResover configuration to foundation-vertx, and use it in service-registry and config-cc to support dynamic config by user

### PR DESCRIPTION
**Pull request by jeho0815**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    ArchaiusUtils.setProperty("addressResolver.rdFlag", true);
+    ArchaiusUtils.setProperty("addressResolver.searchDomains",
+        "default.svc.local.cluster,svc.local.cluster,local.cluster");
+    ArchaiusUtils.setProperty("addressResolver.test.searchDomains",
+        "test.svc.local.cluster,svc.local.cluster,local.cluster");
+    ArchaiusUtils.setProperty("addressResolver.ndots", 3);
+    ArchaiusUtils.setProperty("addressResolver.rotateServers", true);
+    AddressResolverOptions aroc = AddressResolverConfig.getAddressResover("test");
+    Assert.assertEquals(2, aroc.getServers().size());
+    Assert.assertEquals("8.8.8.8", aroc.getServers().get(0));
```
Comment (WillemJiang) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: We could use below lines to check the list

  assertThat(aroc.getServers(), is(Arrays.asList("8.8.8.8", "8.8.4.4")));

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package org.apache.servicecomb.foundation.vertx;
```
Comment (WillemJiang) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: Please add the license header.

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java [hunk 3]
```diff
@@ -40,8 +61,7 @@ public void testGetResover() {
     ArchaiusUtils.setProperty("addressResolver.ndots", 3);
     ArchaiusUtils.setProperty("addressResolver.rotateServers", true);
     AddressResolverOptions aroc = AddressResolverConfig.getAddressResover("test");
-    Assert.assertEquals(2, aroc.getServers().size());
-    Assert.assertEquals("8.8.8.8", aroc.getServers().get(0));
+    Assert.assertThat(aroc.getServers(), is(Arrays.asList("8.8.8.8", "8.8.4.4")));
     Assert.assertEquals("test.svc.local.cluster", aroc.getSearchDomains().get(0));
```
Comment (WillemJiang) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: This line can be changed to use assertThat.

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java [hunk 4]
```diff
@@ -40,8 +61,7 @@ public void testGetResover() {
...
     AddressResolverOptions aroc1 = AddressResolverConfig.getAddressResover("test1");
     Assert.assertEquals("default.svc.local.cluster", aroc1.getSearchDomains().get(0));
```
Comment (WillemJiang) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: Please use assertThat here.

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java [hunk 5]
```diff
@@ -40,8 +61,7 @@ public void testGetResover() {
     ArchaiusUtils.setProperty("addressResolver.ndots", 3);
     ArchaiusUtils.setProperty("addressResolver.rotateServers", true);
     AddressResolverOptions aroc = AddressResolverConfig.getAddressResover("test");
-    Assert.assertEquals(2, aroc.getServers().size());
-    Assert.assertEquals("8.8.8.8", aroc.getServers().get(0));
+    Assert.assertThat(aroc.getServers(), is(Arrays.asList("8.8.8.8", "8.8.4.4")));
     Assert.assertEquals("test.svc.local.cluster", aroc.getSearchDomains().get(0));
     AddressResolverOptions aroc1 = AddressResolverConfig.getAddressResover("test1");
```
Comment (WillemJiang) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: Please use assertThat here.

Comment (jeho0815) [reply foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: ok

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java (hunk 4 above)
Comment (jeho0815) [reply foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: ok

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java (hunk 3 above)
Comment (jeho0815) [reply foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/TestAddressResoverConfig.java]: ok

**Pull request merged**